probe/process/walker.go

```diff
@@ -8,19 +8,22 @@ type Process struct {
 	Comm      string
 	Cmdline   string
 	Threads   int
+	Jiffies   uint64
+	RSSBytes  uint64
 }
 
 // Walker is something that walks the /proc directory
 type Walker interface {
-	Walk(func(Process)) error
+	Walk(func(Process, Process)) error
 }
 
 // CachingWalker is a walker than caches a copy of the output from another
 // Walker, and then allows other concurrent readers to Walk that copy.
 type CachingWalker struct {
-	cache     []Process
-	cacheLock sync.RWMutex
-	source    Walker
+	cache         []Process
+	previousByPID map[int]Process
+	cacheLock     sync.RWMutex
+	source        Walker
```

Comment:
If we don't need guaranteed ordering of `Walk(...)`, it *might* be cleaner to change the type of `cache` to match `previousByPID`, so processes are always indexed by pid.